src/test/java/FileUtilityTest.java

```diff
@@ -243,4 +280,553 @@ public void testSetPermissions() throws IOException {
+        expectedPathsToMonitor.forEach(expectedPath -> {
+            WatchEvent.Kind<?>[] actualEvents = new WatchEvent.Kind<?>[actualPathToEventKindMapCopy.get(expectedPath).size()];
+            actualPathToEventKindMapCopy.get(expectedPath)
+                    .toArray(actualEvents);
+            WatchEvent.Kind<?>[] expectedEvents = new WatchEvent.Kind<?>[actualPathToEventKindMapCopy.get(expectedPath).size()];
+            actualPathToEventKindMapCopy.get(expectedPath)
+                    .toArray(expectedEvents);
+            Assertions.assertArrayEquals(expectedEvents, actualEvents, "Expected and actual events don't match for path - " + expectedPath);
+        });
+    }
+
+    /**
+     * Test unregistering a directory.
+     * @param pathToRegisterString Path to register
+     * @param pathToDeRegisterString Sub path of the pathToRegisterString
+     * @param maxDepth  Maximum children depth from specified path to register and trigger events
+     * @throws IOException Thrown if registering a WatchService or walking through a path  or modifying it fails
+     * @throws InterruptedException Thrown if the current thread is interrupted
+     */
+    @ParameterizedTest
+    @CsvSource(
+        {
+            "fold4, fold4/fold4_child, 3", // existing directory to register and path to deregister
+            ", fold3, 3" // null with existing directory
+        }
+    )
+    public void testFolderWatchServiceDeRegistering(String pathToRegisterString, String pathToDeRegisterString, int maxDepth) throws IOException, InterruptedException {
+        // Test set up
+        createTestBed();
+
+        if(pathToRegisterString != null) {
+            Path pathToRegister = testBedPath.resolve(pathToRegisterString);
+            actualPathToEventKindMap.clear();
+            FileUtility.registerWatchServiceIfPathIsADirectoryAndIsNotAlreadyRegistered(
+                    pathToRegister,
+                    maxDepth,
+                    (path, watchEvent) -> {
+                        logger.info("CONSUMER: Event kind:" + watchEvent.kind() + ". File affected: " + path + ".");
+                        String absolutePathString = path.toAbsolutePath().toString();
+                        actualPathToEventKindMap.putIfAbsent(absolutePathString, new LinkedList<>());
+                        actualPathToEventKindMap.get(absolutePathString)
+                                .add(watchEvent.kind());
+
+                    },
+                    StandardWatchEventKinds.ENTRY_CREATE,
+                    StandardWatchEventKinds.ENTRY_DELETE,
+                    StandardWatchEventKinds.ENTRY_MODIFY,
+                    StandardWatchEventKinds.OVERFLOW
+            );
+
+            // Perform actions to trigger the events
+            Path createdFilePath = pathToRegister.resolve("parent_random.txt");
+            Files.createFile(createdFilePath);
+            Thread.sleep(1_000);
```

Comment:
Can we put this as an entry into test.properties?

Reply:
done